rust/hyperlane-base/src/db/rocks/hyperlane_db.rs

```diff
@@ -23,6 +23,7 @@ const MESSAGE_DISPATCHED_BLOCK_NUMBER: &str = "message_dispatched_block_number_"
 const MESSAGE: &str = "message_";
 const NONCE_PROCESSED: &str = "nonce_processed_";
 const GAS_PAYMENT_BY_SEQUENCE: &str = "gas_payment_by_sequence_";
+const HIGHEST_PROCESSED_MESSAGE_NONCE: &str = "highest_processed_message_nonce_";
```

Comment:
a little worried this will get confused as meaning the highest nonce of a message that has actually been processed on chain. Maybe renaming this and retrieve_highest_processed_message_nonce to be "seen" message would help?